Add BxDecay0 docs to the user manual (#396)
legend-exp/remage: docs/manual/generators.md

To generate double beta decay physics we interface with the _bxdecay0_ package.
This requires _remage_ to be build with _bxdecay0_ support see {ref}`install`.
The main documentation for this extension is available
[here](https://github.com/BxCppDev/bxdecay0).

The macro [command](project:../rmg-commands.md#rmggeneratorselect):

`/RMG/Generator/Select BxDecay0`

> can be used to select the Decay0 generator.

Reviewer: ```suggestion
can be used to select the BxDecay0 generator.
```

_remage_ adds two macro commands to control the generator, which should cover
all required use cases. While the original _bxdecay0_ double beta decay
generator macro commands are still available to the user, the _remage_ commands
offer a simpler and more intuitive alternative.

The macro command to generate double beta decay physics
<project:../rmg-commands.md#rmggeneratorbxdecay0dbd> expects the isotope name,

> the name of the process you want to simulate (like 2vbb or 0vbb) and an optional

Reviewer: ```suggestion
the name of the process you want to simulate (like `2vbb` or `0vbb`) and an optional
```

> energy level for the daughter nucleus. All available isotopes and processes are

Reviewer: ```suggestion
level for the daughter nucleus. All available isotopes and processes are
```

> listed [here](project:../rmg-commands.md#rmggeneratorbxdecay0dbd). The energy
> level defaults to zero and can therefore be omitted in most cases.

Reviewer: ```suggestion
listed [here](project:../rmg-commands.md#rmggeneratorbxdecay0dbd). The nuclear
level defaults to ground state and can therefore be omitted in most cases.
```

For example we can generate two-neutrino double beta decay to the $0^+$ ground
state of $^{76}$Ge with:

```
/RMG/Generator/Select BxDecay0
/RMG/Generator/BxDecay0/dbd Ge76 2vbb
```

> The second macro command is used to generate background physics with _bxdecay0_.

Reviewer: "background" is a nebulous name, why not explicitly say "radioactive decay" or so?

The command <project:../rmg-commands.md#rmggeneratorbxdecay0background> takes
only a single string as parameter, which should be the name of the isotope.
Again all isotope candidates are listed
[here](project:../rmg-commands.md#rmggeneratorbxdecay0background). Using these
we can for example simulate a $^{60}$Co background source:

```
/RMG/Generator/Select BxDecay0
/RMG/Generator/BxDecay0/background Co60
```

The current seed of Geant4 will be forwarded to _bxdecay0_. This means that if
any seed is specified using the
<project:../rmg-commands.md#rmgmanagerrandomization> commands, that seed will be
forwarded to _bxdecay0_.

> :::{note}

Reviewer: ```suggestion
:::{warning}
```
